README.md

```diff
@@ -1,2 +1,96 @@
-# module-slack
-A repository to put the ballerina slack connector source code
+## Module Overview
+
+The `ballerinax/slack` module provides a slack client that allows you to access the Slack API through Ballerina.
+
+The following sections provide you details on how to use the slack connector.
+
+- [Compatibility](#compatibility)
+- [Feature Overview](#feature-overview)
+- [Getting Started](#getting-started)
+- [Samples](#samples)
+
+## Compatibility
+
+|                             |           Version           |
+|:---------------------------:|:---------------------------:|
+| Ballerina Language          |            1.2.0            |
+
+## Feature Overview
+
+### Slack Clients
+There are basically 5 clients provided by Ballerina to interact with different API groups in Slack Web API. 
+1. **slack:Client** - This client is the top most client in the slack module. This can be used to get the relevant client associated with the operation
```

Comment:
```suggestion
1. **slack:Client** - This client is the top-most client in the Slack module. This can be used to get the relevant client associated with the operation
```